pkg/logs/input/listener/tcp.go

```diff
@@ -32,9 +31,8 @@ func NewTCPListener(pp pipeline.Provider, source *config.LogSource) (*TCPListene
 		return nil, err
 	}
 	source.Status.Success()
-	connHandler := NewConnectionHandler(pp, source)
+	connHandler := NewConnectionHandler(pp, source, func() (net.Conn, error) { return nil, fmt.Errorf("dropping connection") })
```

Comment:
could we add a small comment on why we use this function that returns an error?

Reply:
Agreed, I changed the logic though